File: app/src/main/java/com/dicoding/moviecataloguerv/data/source/local/database/AppDatabase.java
package com.dicoding.moviecataloguerv.data.source.local.database;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

import com.dicoding.moviecataloguerv.data.source.model.Movie;
import com.dicoding.moviecataloguerv.data.source.model.Search;
import com.dicoding.moviecataloguerv.data.source.model.TvShow;


@Database(entities = {Movie.class, TvShow.class, Search.class}, version = 1)
public abstract class AppDatabase extends RoomDatabase {

    public abstract MovieDao movieDao();

    public abstract TvShowDao tvShowDao();

    public abstract SearchDao searchDao();

    private static volatile AppDatabase instance = null;

    @NonNull
    public static synchronized AppDatabase getInstance(Context context) {
        if (instance == null) {
            synchronized (AppDatabase.class) {
                if (instance == null) {
                    instance = Room.databaseBuilder(context.getApplicationContext(),
                            AppDatabase.class, "movie_catalogue_db")
                            .fallbackToDestructiveMigration()
                            .build();
                }
            }
        }
        return instance;
    }
}

File: core/src/main/kotlin/com/ryanrvldo/core/data/source/local/database/AppDatabase.kt
package com.ryanrvldo.core.data.source.local.database

import androidx.room.Database
import androidx.room.RoomDatabase
import com.ryanrvldo.core.data.source.local.database.dao.MovieFavoriteDao
import com.ryanrvldo.core.data.source.local.database.dao.SearchHistoryDao
import com.ryanrvldo.core.data.source.local.database.dao.TvShowFavoriteDao
import com.ryanrvldo.core.data.source.local.entity.MovieFavoriteEntity
import com.ryanrvldo.core.data.source.local.entity.SearchHistoryEntity
import com.ryanrvldo.core.data.source.local.entity.TvShowFavoriteEntity

@Database(
    entities = [MovieFavoriteEntity::class, TvShowFavoriteEntity::class, SearchHistoryEntity::class],
    version = 1,
    exportSchema = false
)
abstract class AppDatabase : RoomDatabase() {
    abstract fun movieFavoriteDao(): MovieFavoriteDao
    abstract fun tvShowFavoriteDao(): TvShowFavoriteDao
    abstract fun searchHistoryDao(): SearchHistoryDao
}
